src/gen.ts

```diff
@@ -429,14 +448,22 @@ export class Generator {
             ruleParseFns: this.writeAllRuleParseFns(this.expandedGram),
             parseResult: this.writeParseResultClass(this.expandedGram),
             usesEOF: usesEOF(this.expandedGram),
+            eofType: this.unionEnums ? this.astKindsType("$EOF") : `ASTKinds.$EOF`,
             includeGrammar: this.includeGrammar,
         });
         return writeBlock(parseBlock).join("\n");
     }
+
+    private astKindsType(name: string): string {
+        if (this.unionEnums) {
+            return this.astKindsByName.get(name) ?? "never"; 
```

Comment:
I'm assuming that we don't expect the `?? "never"` case to happen? It's just satisfying the type checker?

Reply:
Yes it shouldn't happen, the map should contain a value for the name (or it's a bug). But I like a logical default better than just a non-null assertion `return this.astKindsByName.get(name)!`